packages/@aws-cdk/core/lib/bundling.ts

```diff
@@ -158,33 +158,7 @@ export class BundlingDockerImage {
    * @deprecated use DockerImage.fromBuild()
    */
   public static fromAsset(path: string, options: DockerBuildOptions = {}) {
-    const buildArgs = options.buildArgs || {};
-
-    if (options.file && isAbsolute(options.file)) {
-      throw new Error(`"file" must be relative to the docker build directory. Got ${options.file}`);
-    }
-
-    // Image tag derived from path and build options
-    const input = JSON.stringify({ path, ...options });
-    const tagHash = crypto.createHash('sha256').update(input).digest('hex');
-    const tag = `cdk-${tagHash}`;
-
-    const dockerArgs: string[] = [
-      'build', '-t', tag,
-      ...(options.file ? ['-f', join(path, options.file)] : []),
-      ...flatten(Object.entries(buildArgs).map(([k, v]) => ['--build-arg', `${k}=${v}`])),
-      path,
-    ];
-
-    dockerExec(dockerArgs);
-
-    // Fingerprints the directory containing the Dockerfile we're building and
-    // differentiates the fingerprint based on build arguments. We do this so
-    // we can provide a stable image hash. Otherwise, the image ID will be
-    // different every time the Docker layer cache is cleared, due primarily to
-    // timestamps.
-    const hash = FileSystem.fingerprint(path, { extraHash: JSON.stringify(options) });
-    return new BundlingDockerImage(tag, hash);
+    DockerImage.fromBuild(path, options) as BundlingDockerImage;
```

Comment:
What's doing on here? Seems like the `return` statement is missing and also the downcast seems hazardous.

 @jogold any comments?

Reply:
The original API returned `BundlingDockerImage`, hence returning the same to keep it backwards compat.

The alternative is to mark the API return type in the method signature and let typescript do the downcast.

Reply:
I am confused. This code does not return anything (there is no `return` statement). I don't see how this alternative can safely work because `fromBuild` does not return a `BuildingDockerImage`, is it?

Reply:
Oops, it should have a `return` statement. That's a miss. I'll submit a fix.

`fromBuild()` returns a `DockerImage` which is a subtype of `BundlingDockerImage`, so downcasting should be fine, no?